src/Microsoft.FeatureManagement/VariantAllocation/Group.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using System.Collections.Generic;
+
+namespace Microsoft.FeatureManagement.VariantAllocation
+{
+    /// <summary>
+    /// The definition of a group allocation.
+    /// </summary>
+    public class Group
```

Comment:
It has a bit more than what I'd expect a type named `Group` to have. Perhaps `GroupAllocation`? Would match with the type summary.

Same with User and Percentile.